### ISSUE TITLE
Crash on startup with directory (but works on re-attempt)

### ISSUE THREAD

**Issue by mcackay** [phaibin/MoliView#28]
Version 1.8.0

I've spotted a semi-reproducible crash, while trying to launch Moliview and open a directory.

I have an Automator service -- with the only action "Open Finder Items"  ... Open with "MoliView.app", with a shortcut bound to it.

* With MoliView not running, and after the computer has been idling for a while, I run this service on a highlighted directory ... it will often crash. 
* If I repeat the Automator task again, it works right away.

I was able to reproduce this a couple of times after a minute or so of idling, which is how I got the console logs below. But it's not consistent -- currently it's loading fine.

(I'm yet to reproduce it by dropping the directory onto the Dock icon. Will let you know if I can confirm that too).

Console logs showing an error:
```
<snip>
default	16:44:08.147538+1200	MoliView	<PKDiscoveryDriver:0x14b666c00> installing watchers for continuous discovery
default	16:44:08.147660+1200	MoliView	networkd_settings_read_from_file initialized networkd settings by reading plist directly
default	16:44:08.169164+1200	MoliView	[0x14b2057e0] invalidated after the last release of the connection object
default	16:44:08.174494+1200	MoliView	0x1210370c0 - [PID=0] WebProcessCache::setApplicationIsActive: (isActive=1)
default	16:44:08.175033+1200	MoliView	0x121031000 - NetworkProcessProxy::sendXPCEndpointToProcess(0x121019200) state = 0 has connection = 0 XPC endpoint message = 0x14b3040e0
default	16:44:08.175101+1200	MoliView	0x121031000 - NetworkProcessProxy::sendXPCEndpointToProcess(0x1210681d0) state = 0 has connection = 0 XPC endpoint message = 0x14b3040e0
default	16:44:08.194885+1200	MoliView	Initializing connection
default	16:44:08.194914+1200	MoliView	Removing all cached process handles
default	16:44:08.194931+1200	MoliView	Sending handshake request attempt #1 to server
default	16:44:08.194940+1200	MoliView	Creating connection to com.apple.runningboard
default	16:44:08.194948+1200	MoliView	[0x14b665e20] activating connection: mach=true listener=false peer=false name=com.apple.runningboard
default	16:44:08.204577+1200	MoliView	Handshake succeeded
default	16:44:08.204598+1200	MoliView	Identity resolved as app<application.com.phaibin.iView.17790258.17790263(501)>
default	16:44:08.223663+1200	MoliView	[0x139fe9af0] activating connection: mach=false listener=false peer=false name=com.apple.ViewBridgeAuxiliary
default	16:44:08.223866+1200	MoliView	[0x139f7a6c0] activating connection: mach=false listener=false peer=false name=(anonymous)
default	16:44:08.224008+1200	MoliView	[0x14b6e8b00] activating connection: mach=false listener=true peer=false name=(anonymous)
default	16:44:08.224018+1200	MoliView	[0x14b6e8b00] Channel could not return listener port.
default	16:44:08.224256+1200	MoliView	[0x139fe9af0] invalidated after the last release of the connection object
error	16:44:08.226079+1200	MoliView	cannot open file at line 49437 of [1b37c146ee]
error	16:44:08.226125+1200	MoliView	os_unix.c:49437: (2) open(/private/var/db/DetachedSignatures) - No such file or directory
default	16:44:08.227227+1200	MoliView	[0x14b7e6ba0] activating connection: mach=true listener=false peer=false name=com.apple.scopedbookmarksagent.xpc
default	16:44:08.235625+1200	MoliView	[0x14b7e6ba0] invalidated after the last release of the connection object
default	16:44:08.256453+1200	MoliView	StorageAccessPromptQuirkController::setCachedQuirks: Loaded 1 storage access prompt(s) quirks from WebPrivacy.
default	16:44:08.256499+1200	MoliView	0x12100ca50 - [PID=71075, throttler=0x121068298] ProcessThrottler::Activity::Activity: Starting foreground activity / 'Lifetime Activity'
default	16:44:08.256551+1200	MoliView	0x121068298 - [PID=71075] ProcessThrottler::didConnectToProcess
default	16:44:08.256558+1200	MoliView	0x12103e430 - ProcessAssertion::acquireSync Trying to take RBS assertion 'Jetsam Boost' for process with PID=71075
default	16:44:08.256566+1200	MoliView	0x121068298 - [PID=71075] ProcessThrottler::setThrottleState: Updating process assertion type to 3 (foregroundActivities=1, backgroundActivities=0)
default	16:44:08.256582+1200	MoliView	0x121031000 - NetworkProcessProxy::sendXPCEndpointToProcess(0x1210681d0) state = 1 has connection = 1 XPC endpoint message = 0x14b3040e0
def
<snip>
```

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by mcackay** [comment]
## WARNING: POSSIBLE MALWARE LINK ^^^
Have reported user and post to to Gitlab.

**Comment by mcackay** [comment]
OK -- so some updates:

* Managed to reproduce this using drag-onto-icon with a folder.
* I hadn't tried before, but it's also happening when opening a single image
.
* It _mostly_ works on second try ... but I did end up with a period where I couldn't open up any file in Moliview launched from the Finder. The console logs were displaying basically the same error lines.
* However, if I opened Moliview first from the Doc, then dragged a Folder or an Image onto the app window, then it loaded fine.
* ... coming back after being afk for a few hours afk, it's return to the 'crash once, launch on second' behaviour.

**Comment by phaibin** [comment]
This is also related to macOS 15.0😂. Wish Apple would release macOS 15.0 ASAP🙏. So I can use the latest Xcode to build the project. Then the App will behave the same in 14.0 and 15.0.

**Comment by phaibin** [comment]
@mcackay Please check if the bug has been fixed in 1.9.0.

**Comment by mcackay** [comment]
Yes, seems fine so far. Thanks again.
